plugins/optimization-detective/detect.js

```diff
@@ -490,13 +491,17 @@ export default async function detect( {
 		} );
 	}
 
-	/** @type {LCPMetric[]} */
+	/** @type {(LCPMetric|LCPMetricWithAttribution)[]} */
 	const lcpMetricCandidates = [];
 
 	// Obtain at least one LCP candidate. More may be reported before the page finishes loading.
 	await new Promise( ( resolve ) => {
 		onLCP(
-			( /** @type LCPMetric */ metric ) => {
+			/**
+			 *
+			 * @param {LCPMetric|LCPMetricWithAttribution} metric
+			 */
+			( metric ) => {
```

Comment:
Could this go back to being defined inline? The type is picked up for me in PhpStorm:

![image](https://github.com/user-attachments/assets/62373e51-bc77-45e9-9eaf-70445769237b)


```suggestion
			( /** @type {LCPMetric|LCPMetricWithAttribution} */ metric ) => {
```

Reply:
But your PhpStorm clearly only sees `LCPMetric`, not `LCPMetricWithAttribution`

Reply:
Good point! 

What's strange to me is why this typing is even needed. 

With 3614c137afeff9275bbb1c5390d22e588fcd8280 I added more typing for the attribution report functions. Then PHPStorm reports that the arg is detected as a `LCPMetricWithAttribution`:

![image](https://github.com/user-attachments/assets/42636d99-3d3f-42e9-80cb-0333860370e2)

But TypeScript is giving a warning about it being implicit `any`:

![image](https://github.com/user-attachments/assets/ace4474d-b52d-42e1-bf7d-bf6e3e33942f)

Nevertheless, `onLCP` is defined as being either `OnLCPFunction` or `OnLCPWithAttributionFunction`:

![image](https://github.com/user-attachments/assets/34edd326-9060-4541-8762-64f05639ff44)

So I would have thought that this would be later be reflected in the type of the arg here. But instead it is showing up as `any`, even though the common `ReportOpts` arg is carried through:

![image](https://github.com/user-attachments/assets/05bd09c4-a2e9-48c3-b692-89d515b30e90)